src/MultiPartStreamFactory.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Http\Message;
+
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Factory for creating Multipart Streams.
+ *
+ * @author Robert Schönthal <robert.schoenthal@gmail.com>
+ */
+interface MultiPartStreamFactory extends StreamFactory
+{
+    /**
+     * Creates a Multipart Stream from parameters and/or files.
+     *
+     * @param array $parameters
+     * @param array $files
```

Comment:
mh, my thought were a list of files/paths. but yeah a list of streams would be more consistent. but i dont it would be good to use streams:

* Guzzle uses LazyOpenStream for files. 
* Diactores something different
* we need to verify that this stream is a file?
* how to pass extra information like the filename (as array key)?

i dont know exactly what would be best. Streams would be consistent, but i dont know if that is flexible enough